### ISSUE TITLE
[Datepicker] Fix behavior of MonthYearSelect and NextMonthButton with multiple months 

### ISSUE THREAD

**Issue by choyongjoon** [uber/baseweb#2128]
In Calendar with multiple months, the MonthYearSelect and NextMonthButton of 2nd month is not working correctly. 

MonthYearSelect: If October is selected in the 2nd month, the 2nd month will be November. And the 1st month will be October

NextMonthButton: If NextMonthButton is clicked, the months are increased by 2

And when `orientation={ORIENTATION.horizontal}` is applied, the MonthYearSelect is not shown. But without orientation prop, MonthYearSelect is shown. So I think the author decided to not support MonthYearSelect for multiple months and hide it but forgot to handle when orientation is undefined.

How about to support MonthYearSelect with multiple months too? It's not that hard and users who didn't set oritentation prop alerady see MonthYearSelect.

https://codesandbox.io/s/calendar-with-multiple-months-1cn1l
~https://codesandbox.io/s/calendar-with-multiple-months-d3rgw~

## Current Behavior

MonthYearSelect: If August is selected in the 2nd month, the 2nd month will be September. And the 1st month will be August.

![baseui-datepicker-month-select-bug](https://user-images.githubusercontent.com/1563230/67553729-d123c880-f748-11e9-9a63-e398a6874c11.gif)

NextMonthButton: If NextMonthButton is clicked, the months are increased by 2

![baseui-datepicker-next-button-bug](https://user-images.githubusercontent.com/1563230/67553745-d84ad680-f748-11e9-92dc-c32a1d203020.gif)

## Expected Behavior

MonthYearSelect: If August is selected in the 2nd month, the 2nd month will be August. And the 1st month will be July.

NextMonthButton: If NextMonthButton is clicked, the months are increased by 1

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v9.9.0  |
| React   |         |
| browser |  Firefox 70.0b14 |

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.

**Comment by gergelyke** [comment]
hi @choyongjoon - thanks for opening the issue! based on the description, I am not sure if I fully understand the issue. Any chance you can do a screen recording, and post a gif?

**Comment by chasestarr** [comment]
Ah I see what you mean, styles guides from our designers define that the month titles should not be interactive if more than one month is displayed. In that case, this is a bug, and `orientation = undefined` should operate the same as `orientation = ORIENTATION.horizontal`.

**Comment by chasestarr** [comment]
To reproduce, comment out the `orientation` prop from the codesandbox example linked above

**Comment by choyongjoon** [comment]
@chasestarr You're right. I missed to remove the `orientation` prop and fork the codesandbox example. I modified the link. So I wrote multiple issues about displaying multiple months in here. 1) When `orientation = undefined`, there is a interactive month select. 2) the interactive month select is not working correctly. 3) the next month button is working differently with the prev month button. If 1) is resolved, 2) is not a problem. So 1) should be fixed at first, and 3) should be checked whether it is design decision or a bug.

**Comment by choyongjoon** [comment]
@gergelyke I uploade 2 gifs. Thanks for your comment.

**Comment by foodforarabbit** [comment]
I was looking into this ticket. The "fix" is easy for this particular issue, but I just noticed a few things, not sure how we want to proceed with them

1) vertical seems to not be working at all. It isn't vertical.
2) the prev/next month arrows on each subsequent month seem to have the index added to the amount of months it changes by. So for example, even if we fix the horizontal display issue, in a horizontal calendar with 3 monthsShown, the left arrow on the first month goes back by one month (as expected) while the right arrow on the 3rd month jumps by 3 months. I get how it is supposed to kinda be like "the next three months" but it might be a bit unintuitive. BUT, if that is the correct case, then the left arrow in the first month should jump BACK 3 months. So either way, it needs to be consistent.
3) in vertical mode, it should not have that multi month jumping thing. The left/right arrows work fine in the first month but for all the other months it doesn't seem to make sense. I had to manually add `flex-direction: 'column'` to kinda see what it looks like and it doesn't feel right.

Could I get a link to the style guide? I'm interested to see how it is "supposed" to work. Also feel free to assign this ticket to me thanks!
